pages/timeline.vue

```diff
@@ -0,0 +1,197 @@
+<template>
+    <PageFrameSm class="news">
+        <header>
+            <h1>
+                <span>Timeline</span>
+                <small>Notable Opepen Activity</small>
+            </h1>
+        </header>
+
+        <hr>
+
+        <section v-if="isAdmin">
+            <div class="add-tweet">
+                <input
+                    type="text"
+                    v-model="tweetInput"
+                    placeholder="Add a new tweet URL or Id"
+                />
+
+                <Button class="input-button" @click="addTweet">Add</Button>
+            </div>
+            <p v-if="successMessage" class="success-message">{{ successMessage }}</p>
+            <p v-if="errorMessage" class="error-message">{{ errorMessage }}</p>
+        </section>
+
+        <section class="tweets">
+            <Button @click="toggleSort" class="sort-button">
+                <Icon :type="arrowType" />
+            </Button>
+            <CuratedTweets 
+                :key="refreshKey"
+                :sort="sort"
+            />
+        </section>
+    </PageFrameSm>
+</template>
+
+
+<script setup>
+const tweetInput = ref('')
+const refreshKey = ref(Date.now())
+const successMessage = ref('')
+const errorMessage = ref('')
+
+const sort = ref('-tweet_created_at')
+const arrowType = ref('arrow-down') 
+
+function extractTweetId(url) {
+    const regex = /https?:\/\/(?:www\.)?(?:twitter\.com|x\.com)\/[^\/]+\/status(?:es)?\/(\d+)/i
+    const match = url.match(regex)
+    return match ? match[1] : null
+}
+
+async function addTweet() {
+    successMessage.value = ''
+    errorMessage.value = ''
+
+    if (!tweetInput.value) {
+        errorMessage.value = 'Tweet input is empty.'
+        return
+    }
+
+    let tweetId = tweetInput.value.trim()
+    if (tweetId.includes('twitter.com') || tweetId.includes('x.com')) {
+        const extractedId = extractTweetId(tweetId)
+
+        if (!extractedId) {
+            errorMessage.value = 'Could not extract Tweet ID.'
+            return
+        }
+        tweetId = extractedId
+    }
+
+    try {
+        const { data, error, execute } = await useApiPost('/curated-tweets', {
+            body: JSON.stringify({ ids: [tweetId] }),
```

Comment:
Can't/shouldn't use a composable within a function body. this call belongs in the root of the component;

The `data`, `error`, .. attributes are `ref`s so they're reactive to changes and can be used as is in templates.

That way you don't need all this custom error handling shite in this function either 😅